src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/MetricItemExtensions.cs

```diff
@@ -0,0 +1,322 @@
+// <copyright file="MetricItemExtensions.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Reflection.Emit;
+using System.Runtime.CompilerServices;
+using Google.Protobuf;
+using Google.Protobuf.Collections;
+using OpenTelemetry.Metrics;
+using OtlpCollector = Opentelemetry.Proto.Collector.Metrics.V1;
+using OtlpCommon = Opentelemetry.Proto.Common.V1;
+using OtlpMetrics = Opentelemetry.Proto.Metrics.V1;
+using OtlpResource = Opentelemetry.Proto.Resource.V1;
+
+namespace OpenTelemetry.Exporter.OpenTelemetryProtocol.Implementation
+{
+    internal static class MetricItemExtensions
+    {
+        private static readonly ConcurrentBag<OtlpMetrics.InstrumentationLibraryMetrics> MetricListPool = new ConcurrentBag<OtlpMetrics.InstrumentationLibraryMetrics>();
+        private static readonly Action<RepeatedField<OtlpMetrics.Metric>, int> RepeatedFieldOfMetricSetCountAction = CreateRepeatedFieldOfMetricSetCountAction();
+
+        internal static void AddBatch(
+            this OtlpCollector.ExportMetricsServiceRequest request,
+            OtlpResource.Resource processResource,
+            in MetricItem metricItem)
+        {
+            var metricsByLibrary = new Dictionary<string, OtlpMetrics.InstrumentationLibraryMetrics>();
+            var resourceMetrics = new OtlpMetrics.ResourceMetrics
+            {
+                Resource = processResource,
+            };
+            request.ResourceMetrics.Add(resourceMetrics);
+
+            foreach (var metric in metricItem.Metrics)
+            {
+                var otlpMetric = metric.ToOtlpMetric();
+                if (otlpMetric == null)
+                {
+                    OpenTelemetryProtocolExporterEventSource.Log.CouldNotTranslateMetric(
+                        nameof(MetricItemExtensions),
+                        nameof(AddBatch));
+                    continue;
+                }
+
+                var instrumentationLibraryName = "TODO: We need instrumentation library info";
+                var instrumentationLibraryVersion = "0.0.1";
```

Comment:
@victlu We do not currently have instrumentation library information on either `MetricItem` or `IMetric` - which makes more sense?

Reply:
I'm taking a stab at this in a separate PR. Will push up in a moment...

Reply:
#2124

Reply:
I left comments in #2124 regarding Description. But, I think you are right in we need to pass a reference to the Meter (rather than instrument).